StreamChatClient_v3/WebSocketClient/WebSocketPingController.swift

```diff
@@ -0,0 +1,76 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import Foundation
+
+class WebSocketPingController {
+    typealias PingAction = () -> Void
+    typealias ForceReconnectAction = () -> Void
+    
+    /// The time interval to ping connection to keep it alive.
+    static let pingTimeInterval: TimeInterval = 25
+    static let pongTimeoutTimeInterval: TimeInterval = 3
+    
+    private let timerType: Timer.Type
+    private let timerQueue: DispatchQueue
+    
+    /// The timer used for scheduling `ping` calls
+    private lazy var pingTimerControl: RepeatingTimerControl =
+        timerType.scheduleRepeating(timeInterval: Self.pingTimeInterval, queue: timerQueue) { [weak self] in self?.sendPing() }
+    
+    /// The pong timeout timer.
+    private var pongTimeoutTimer: TimerControl?
+    private let ping: () -> Void
+    private let forceReconnect: () -> Void
```

Comment:
There's no reason why this can't be `"internal" var`

Reply:
It prevents from inappropriate usage and any side effects. I don't think that the behaviour of the closure should be able to change later. Better to create another controller.

Reply:
What exactly is the "inappropriate usage" and "side effect" of setting this callback? I'm not sure I understand.

Reply:
What I wanted to say is that it's only for internal usage. It shouldn't be used outside of PingController.
So this is not a desired flow:
```swift
pingController. forceReconnect()
```